mitmproxy/certs.py

```diff
@@ -502,11 +509,28 @@ def add_cert_file(
         raw = path.read_bytes()
         cert = Cert.from_pem(raw)
         try:
-            key = load_pem_private_key(raw, password=passphrase)
-        except ValueError:
-            key = self.default_privatekey
+            private_key = load_pem_private_key(raw, password=passphrase)
+        except ValueError as e:
+            private_key = self.default_privatekey
+            if cert.public_key() != private_key.public_key():
+                raise ValueError(
+                    f'Unable to find private key in "{path.absolute()}": {e}'
+                ) from e
+        else:
+            if cert.public_key() != private_key.public_key():
+                raise ValueError(
```

Comment:
Do we need to handle this line in a similar fashion? https://github.com/mitmproxy/mitmproxy/blob/0ff08f975e52700c02ab590a9af8e0bd10729902/mitmproxy/certs.py#L419